app/js/components/EscrowList.jsx

```diff
@@ -57,22 +76,37 @@ const EscrowList = (props) => (<Card className="mt-2">
               {getEscrowState(escrow) === escrowStates.expired && escrow.seller === web3.eth.defaultAccount &&
               <Button color="warning" size="sm" block
                       onClick={() => props.cancelEscrow(escrow.escrowId)}>Cancel</Button>}
-
-              {getEscrowState(escrow) !== escrowStates.waiting && escrow.buyer === web3.eth.defaultAccount &&
-              <Rating rating={escrow.rating} rateTransaction={props.rateTransaction}
-                      escrowId={escrow.escrowId}/>}
+              {getEscrowState(escrow) === escrowStates.released && !escrow.arbitration && escrow.buyer === web3.eth.defaultAccount &&
+              <Rating rating={parseInt(escrow.rating, 10)} rateTransaction={props.rateTransaction}
+                        escrowId={escrow.escrowId}/>}
+              {getEscrowState(escrow) === escrowStates.waiting && escrow.buyer === web3.eth.defaultAccount  && <Fragment>
+                <Button color="warning" size="sm" block onClick={() => props.payEscrow(escrow.escrowId)}>Mark as paid</Button>
+                <Button color="warning" size="sm" block onClick={() => props.payEscrowSignature(escrow.escrowId)}>Sign as paid (for relayers)</Button>
+              </Fragment>}
+              {getEscrowState(escrow) === escrowStates.paid && <Fragment>
+                <Button color="warning" size="sm" block onClick={() => props.openCase(escrow.escrowId)}>Open case</Button>
+                <Button color="warning" size="sm" block onClick={() => props.openCaseSignature(escrow.escrowId)}>Open a case signature (for relayers)</Button>
```

Comment:
We should move all the button in another stateless component. Can be done in another PR. 
It's just that that section is becoming quite big and hard to read.